### DISCUSSION TITLE
Entity.remove() not working

### DISCUSSION THREAD

**Discussion by MrGeoTech** [Minestom/Minestom#631] [Help] [answered]
So, I have a falling block entity that should be getting removed by calling `fallingBlock.remove()` and I have even removed the viewers from seeing the entity but yet the falling blocks are still there. Btw, it also changes the `isRemoved()` to true so could it be an issue of not sending the remove packets to the players?

**Comment by TheMode** [comment, marked as answer]
I will need a reproducer.

Alternatively, you can try sending the destroy packet manually.

**Reply by MrGeoTech** [reply]
Here is my code. The `gameManager.updateViewable()` just adds all the players in the game instance to the viewable list of the entity and the rest is modeled after emortals block falling and placing animation in their marathon gamemode.
```
long distanceToFall = 30;

Entity fallingBlock = new Entity(EntityType.FALLING_BLOCK);
FallingBlockMeta fallingBlockMeta = (FallingBlockMeta) fallingBlock.getEntityMeta();
fallingBlockMeta.setBlock(block);
fallingBlock.setVelocity(new Vec(0.0, -distanceToFall - 25, 0.0));
fallingBlock.setNoGravity(true);
gameManager.updateViewable(fallingBlock);
fallingBlock.setInstance(instance, pos.add(0.5, distanceToFall + 1, 0.5));

MinecraftServer.getSchedulerManager().buildTask(() -> {
      instance.setBlock(pos, block);
      System.out.println(fallingBlock.isRemoved());
      fallingBlock.getViewers().forEach(fallingBlock::removeViewer);
      fallingBlock.remove();
      System.out.println(fallingBlock.isRemoved());
}).delay(Duration.ofMillis(distanceToFall * 10)).schedule();
```

**Reply by TheMode** [reply]
This code works for me without the `gameManager.updateViewable(fallingBlock)` line. Why do you need to manually handle the entity viewers in the first place?

**Reply by MrGeoTech** [reply]
Do you not need to add the players so that they can see the falling block? I guess now that I think about it, it's an entity so ig that would make sense. I'll try it.

**Comment by emortaldev** [comment]
This may be related, I have an issue where an AreaEffectCloud hologram that rides a player does not get removed for the player

(in the video, a player sends a message and the hologram is created, 2 seconds later they move instance, then another 3 seconds later the area effect cloud is removed)
https://user-images.githubusercontent.com/28760284/151788197-2afb3173-b4f8-4dfd-a0d4-7791a192f0fc.mp4

When the hologram stops moving, it is removed properly for other players.

**Reply by TheMode** [reply]
Again I will need a reproducer. The video is also not playable.
